src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs

```diff
@@ -148,5 +127,49 @@ private void CalculateHashCode()
                 }
             }
         }
+#else
+        private SqlConnectionPoolKey(SqlConnectionPoolKey key) : base(key)
+        {
+            _credential = key.Credential;
+            _accessToken = key.AccessToken;
+            CalculateHashCode();
+        }
+
+        internal SqlConnectionPoolKey(string connectionString, SqlCredential credential, string accessToken) : base(connectionString)
+        {
+            Debug.Assert(_credential == null || _accessToken == null, "Credential and AccessToken can't have the value at the same time.");
+            _credential = credential;
+            _accessToken = accessToken;
+            CalculateHashCode();
+        }
+
+        public override bool Equals(object obj)
+        {
+            return (obj is SqlConnectionPoolKey key
+                && _credential == key._credential
+                && ConnectionString == key.ConnectionString
+                && string.CompareOrdinal(_accessToken, key._accessToken) == 0);
+        }
+
+        internal void CalculateHashCode()
+        {
+            _hashValue = base.GetHashCode();
+
+            if (_credential != null)
+            {
+                unchecked
+                {
+                    _hashValue = _hashValue * 17 + _credential.GetHashCode();
+                }
+            }
+            else if (_accessToken != null)
+            {
+                unchecked
+                {
+                    _hashValue = _hashValue * 17 + _accessToken.GetHashCode();
+                }
+            }
+        }
+#endif // NETFRAMEWORK
```

Comment:
NIT: I see it's the end of the Netfx **#if-directive**. Instead of this comment I'd suggest wrapping each section in a `#region` and `#endregion` block.